open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java

```diff
@@ -0,0 +1,463 @@
+        logicTable.setGaianNode(sqlResults.getString(gdbNode));
+        logicTable.setLogicalTableName(sqlResults.getString(logicTableName));
+        String def = sqlResults.getString(logicTableDefinition);
+        String[] defs = def.split(", ");//here we need to split with ,+space, it is showed in Gaian LTDEF
+        for (String column : defs) {
+            String[] temp = column.split(" ");
+            defLists.put(temp[0], temp[1]);
+        }
+        logicTable.setLogicalTableDefinition(defLists);
+        return logicTable;
+    }
+
+
+    private Map<String, String> createTableDefinitions(TableContextEvent tableContextEvent, String gaianNodeName, String technicalTableName, String businessTableName, String logicalTableName, List<MappedColumn> mappedColumns){
+        String methodName = "createTableDefinitions";
+        Map<String, String> createdTables = new HashMap();
+        LogicTable backendTable = getMatchingTables(gaianNodeName, Collections.singletonList(logicalTableName));
+        if (backendTable != null) {
+            if (!backendTable.getGaianNode().equals(gaianFrontendName)) {
+                createMirroringLogicalTable(logicalTableName, gaianNodeName);
+            }
+            ConnectorUtils.updateColumnDataType(mappedColumns, backendTable);
+
+            String updatedTable = createTableDefinition(tableContextEvent.getTableSource().getDatabaseSource().getName(), businessTableName, (c -> c.getBusinessName()), mappedColumns, gaianNodeName, logicalTableName);
+            if (updatedTable != null) {
+                createdTables.put(ConnectorUtils.BUSINESS_PREFIX, updatedTable);
+            }
+
+            updatedTable = createTableDefinition(tableContextEvent.getTableSource().getDatabaseSource().getName(), technicalTableName, (c -> c.getTechnicalName()), mappedColumns, gaianNodeName, logicalTableName);
+            if (updatedTable != null) {
+                createdTables.put(ConnectorUtils.TECHNICAL_PREFIX, updatedTable);
+            }
+
+            if (!backendTable.getGaianNode().equals(gaianFrontendName)) {
+                log.info("Remove mirrored logical table: {}", logicalTableName);
+                deleteLogicalTable(logicalTableName);
+            }
+            return createdTables;
+        } else {
+            log.error("error");
+            return null;
+        }
+    }
+
+
+    /**
+     * Set Logical Table mirroring the definition for the given Logical Table Name on another GaianDB node, so its data can be queried remotely.
+     *
+     * @param logicalTableName
+     * @param gaianNodeName
+     */
+    private void createMirroringLogicalTable(String logicalTableName, String gaianNodeName) {
+        log.debug("Set up Logical Table for Gaian node");
+        String setLogicalTableForNode = "call setltfornode('" +
+                logicalTableName + "','" +
+                gaianNodeName + "')";
+
+        executeCustomizedQuery(setLogicalTableForNode);
```

Comment:
A new method calling executeUpdate should be used instead of executeQuery as this statement is not a query, it doesn't return any results.